sdk/auth/shared_key_authorizer.go

```diff
@@ -64,7 +64,10 @@ func (s *SharedKeyAuthorizer) AuxiliaryTokens(_ context.Context, _ *http.Request
 	return []*oauth2.Token{}, nil
 }
 
-// ---
+// ExpireToken has no effect with shared keys
+func (c *SharedKeyAuthorizer) ExpireTokens() error {
+	return fmt.Errorf("SharedKeyAuthorizer tokens cannot expire")
```

Comment:
given the other authorizers are a noop, why not do the same here? perhaps this method wants renaming to `InvalidateTokenCache` or something to better reflect what it's doing?

Reply:
I've done away with this ambiguity and added a `CachingAuthorizer` interface which can be tested for at runtime to call the `InvalidateTokenCache()` method.